pkg/cmd/component/component/add.go

```diff
@@ -0,0 +1,214 @@
+)
+
+// Complete completes AddCompParameters after they've been created.
+func (io *AddCompParameters) Complete(name string, cmd *cobra.Command, args []string) error {
+	if cmd.Flags().NFlag() == 0 || io.Interactive {
+		return initiateInteractiveModeComponent(io, cmd)
+	}
+	return initiateNonInteractiveModeComponent(io)
+}
+
+// Checking the mandatory flags & reusing missingFlagErr in .go
+func checkMandatoryFlags(flags map[string]string) error {
+	missingFlags := []string{}
+	mandatoryFlags := []string{componentNameFlag, applicationNameFlag}
+	for _, flag := range mandatoryFlags {
+		if flags[flag] == "" {
+			missingFlags = append(missingFlags, fmt.Sprintf("%q", flag))
+		}
+	}
+	if len(missingFlags) > 0 {
+		return missingFlagErr(missingFlags)
+	}
+	return nil
+}
+
+func missingFlagErr(flags []string) error {
+	return fmt.Errorf("required flag(s) %s not set", strings.Join(flags, ", "))
+}
+
+func initiateNonInteractiveModeComponent(io *AddCompParameters) error {
+	mandatoryFlags := map[string]string{componentNameFlag: io.ComponentName, applicationNameFlag: io.ApplicationName}
+	if err := checkMandatoryFlags(mandatoryFlags); err != nil {
+		return err
+	}
+	err := ui.ValidateName(io.ApplicationName)
+	if err != nil {
+		return err
+	}
+	err = ui.ValidateName(io.ComponentName)
+	if err != nil {
+		return err
+	}
+	if io.OutputFolder == "./" {
+		path, _ := os.Getwd()
+		io.OutputFolder = path
+	}
+	exists, _ := ioutils.IsExisting(ioutils.NewFilesystem(), io.OutputFolder)
+	if !exists {
+		return fmt.Errorf("the given Path : %s  doesn't exists ", io.OutputFolder)
+	}
+	exists, _ = ioutils.IsExisting(ioutils.NewFilesystem(), filepath.Join(io.OutputFolder, io.ApplicationName))
+	if !exists {
+		return fmt.Errorf("the given Application: %s  doesn't exists in the Path: %s", io.ApplicationName, io.OutputFolder)
+	}
+	exists, _ = ioutils.IsExisting(ioutils.NewFilesystem(), filepath.Join(io.OutputFolder, io.ApplicationName, "components", io.ComponentName))
+	if exists {
+		return fmt.Errorf("the Component : %s  already exists in Path %s", io.ComponentName, filepath.Join(io.OutputFolder, io.ApplicationName, "components", io.ComponentName))
+	}
+
+	return nil
+}
+
+func initiateInteractiveModeComponent(io *AddCompParameters, cmd *cobra.Command) error {
+	log.Progressf("\nStarting interactive prompt\n")
+	promp := !ui.UseDefaultValuesComponent()
+	if promp || cmd.Flag("output").Changed {
+		// ask for output folder
+		io.OutputFolder = ui.AddOutputPath()
+	}
+	if io.OutputFolder != "" {
+		// Check for the path wether it is valid or not
+		exists, _ := ioutils.IsExisting(ioutils.NewFilesystem(), io.OutputFolder)
+
+		if !exists {
+			log.Progressf("the provided Path doesn't exists in you directory : %s", io.OutputFolder)
+			io.OutputFolder = ui.AddOutputPath()
+			// ask for output folder
+		}
+	}
+	if io.OutputFolder == "./" {
+		path, _ := os.Getwd()
+		io.OutputFolder = path
+	}
+	ui.PathGiven = io.OutputFolder
+	if io.ApplicationName != "" {
```

Comment:
Hi @JainamShah-08 , I think all of this can be simplified.  eg. the validation can be done in the prompt/survey input, where you provide validation.

Reply:
Resolved as we discussed in our meeting.